crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -277,10 +278,13 @@ impl Instruction {
                 let array = dfg.get_array_constant(*array);
                 let index = dfg.get_numeric_constant(*index);
 
+                // If we are getting a value from a constant array with a constant index
+                // Then we can substitute that value in place of the array_get instruction.
+                // TODO: what is a non-constant array?
                 if let (Some((array, _)), Some(index)) = (array, index) {
                     let index =
                         index.try_to_u64().expect("Expected array index to fit in u64") as usize;
-
+                    // TODO: Where do we check for Out of Bounds(OOB) error?
```

Comment:
We only check for OOB during acir-gen. Any checks that are known to be out of bounds during SSA are caught right here and just left in the code for acir-gen to find. It doesn't particularly matter where we stop to report these errors as long as that spot eventually has location information for them.